# Bonus: Temperature Analysis I

In [60]:
import pandas as pd
from datetime import datetime as dt

In [61]:
# "tobs" is "temperature observations"
df = pd.read_csv('./Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [62]:
# Convert the date column format from string to datetime
df.date=pd.to_datetime(df.date,infer_datetime_format=True)

In [63]:
# Set the date column as the DataFrame index

df = df.set_index(df['date'])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [64]:
# Drop the date column
df=df.drop(columns='date')
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [65]:
from scipy import stats

In [66]:
# Filter data for desired months
#filter June data
juneData = df[df.index.month==6]
juneData.head()

station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77

In [67]:
# Filter data for desired months
#filter December data
decData = df[df.index.month==12]
decData.head()

station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64

In [68]:
# Identify the average temperature for June
juneData.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [69]:
# Identify the average temperature for December
decData.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [70]:
# Create collections of temperature data
#June Collection
juneTemp = juneData.tobs
juneTemp

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [71]:
# Create collections of temperature data
#December Collection
decTemp = decData.tobs
decTemp

date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [72]:
# Run paired t-test
stats.ttest_ind(juneTemp,decTemp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The mean temperature difference between June and December is 3.9 F which is not much. The t-test with an low p-value indicates that the difference is statistically significant. Eventhough the difference is meaningful, the actual difference is not thereby indicating that you can travel to Hawaii and enjoy 70 degree temperature whole year